r/R/dplyr-functions.R

```diff
@@ -651,3 +651,47 @@ nse_funcs$log <- function(x, base = exp(1)) {
 }
 
 nse_funcs$logb <- nse_funcs$log
+
+nse_funcs$case_when <- function(...) {
+  formulas <- list2(...)
+  n <- length(formulas)
+  if (n == 0) {
+    abort("No cases provided in case_when()")
+  }
+  query <- vector("list", n)
+  value <- vector("list", n)
+  mask <- caller_env()
+  for (i in seq_len(n)) {
+    f <- formulas[[i]]
+    if (!inherits(f, "formula")) {
+      abort("Each argument to case_when() must be a two-sided formula")
+    }
+    query[[i]] <- arrow_eval(f[[2]], mask)
+    value[[i]] <- arrow_eval(f[[3]], mask)
+    if (!nse_funcs$is.logical(query[[i]])) {
+      abort("Left side of each formula in case_when() must be a logical expression")
+    }
+    # TODO: remove these checks after the case_when kernel supports variable-width
+    # types (ARROW-13222)
```

Comment:
Can we just let the Arrow C++ library error out if it doesn't support the types? Is our error message that much better? Seems like extra work to try to keep this validation in sync with the C++ library.

Reply:
Yep, agreed, there's really not much (any?) benefit to having these checks here. Removed in 2ee3c8c5909857452d89a722459127ddde723c05.